### PR TITLE
Added initial react refs guidelines

### PR DESCRIPTION
**Pull request by lededje**

### CODE REVIEW

File: _guidelines/react.md [hunk 1]
```diff
@@ -64,3 +64,19 @@ Avoid using an array index as `key` prop, prefer a unique ID: using the index as
   />
 ))}
 ```
+
+## Refs
+
+Always use ref callbacks. eslint: [`react/no-string-refs`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-string-refs.md)
```
Comment (kapowaz) [inline _guidelines/react.md]: Footnote syntax?

Comment (martyphee) [inline _guidelines/react.md]: Why?  Is it because of performance?

Comment (lededje) [inline _guidelines/react.md]: Deprecated

Comment (martyphee) [inline _guidelines/react.md]: Good to know!  Guess I have a couple hygiene tickets to create.

Comment (lededje) [inline _guidelines/react.md]: Sorry, I was wrong:

"Although string refs are not deprecated, they are considered legacy, and will likely be deprecated at some point in the future. Callback refs are preferred." - https://facebook.github.io/react/docs/more-about-refs.html

Comment (martyphee) [inline _guidelines/react.md]: It would be better to set the standard now though and remove them.

**Comment by mezis** [comment]
@lededje this is shippable, right?

**Comment by lededje** [comment]
Oh yeah, 27 is the new 7

**Comment by lededje** [comment]
Okay I'll stop now ;D

**Pull request merged**